infra/Pos/Communication/PeerState.hs

```diff
@@ -60,34 +59,36 @@ peerStateFromSnapshot (PeerStateSnapshot snapshot) = do
 
 data PeerStateTag
 
--- | Wrapper for monadic action which brings 'NodePeerState'.
-data PeerStateRedirectTag
-
-type PeerStateRedirect =
-    Ether.TaggedTrans PeerStateRedirectTag IdentityT
+type PeerStateRealMonad m =
+    ( MonadIO m, Mockable SharedAtomic m
+    , Ether.MonadReader PeerStateTag (PeerStateCtx m) m )
 
-runPeerStateRedirect :: PeerStateRedirect m a -> m a
-runPeerStateRedirect = coerce
+getPeerStateReal
+    :: PeerStateRealMonad m
+    => NodeId -> m (SharedAtomicT m PeerState)
+getPeerStateReal nodeId = Ether.ask @PeerStateTag >>= \m -> do
+    mV <- atomically $ nodeId `STM.lookup` m
+    case mV of
+      Just v -> return v
+      _ -> do
+        st <- newSharedAtomic def
+        atomically $ do
+            mV' <- nodeId `STM.lookup` m
+            case mV' of
+              Just v -> return v
+              _      -> STM.insert st nodeId m $> st
 
-instance
-    ( MonadIO m, Mockable SharedAtomic m
-    , Ether.MonadReader PeerStateTag (PeerStateCtx m) m
-    , t ~ IdentityT
-    ) => WithPeerState (Ether.TaggedTrans PeerStateRedirectTag t m)
-  where
-    getPeerState nodeId = Ether.ask @PeerStateTag >>= \m -> do
-          mV <- atomically $ nodeId `STM.lookup` m
-          case mV of
-            Just v -> return v
-            _ -> do
-              st <- newSharedAtomic def
-              atomically $ do
-                  mV' <- nodeId `STM.lookup` m
-                  case mV' of
-                    Just v -> return v
-                    _      -> STM.insert st nodeId m $> st
+clearPeerStateReal
+    :: PeerStateRealMonad m
+    => NodeId -> m ()
+clearPeerStateReal nodeId = do
+    m <- Ether.ask @PeerStateTag
+    atomically $ nodeId `STM.delete` m
 
-    clearPeerState nodeId = Ether.ask @PeerStateTag >>= \m -> atomically $ nodeId `STM.delete` m
-    getAllStates = Ether.ask @PeerStateTag >>= \m -> do
-        stream <- atomically $ LT.toList $ STM.stream m
-        PeerStateSnapshot <$> forM stream (mapM readSharedAtomic)
+getAllStatesReal
+    :: PeerStateRealMonad m
+    => m PeerStateSnapshot
+getAllStatesReal = do
```

Comment:
And here it can be `getAllStatesDefault`, for example. Because that's just the simplest possible implementation I guess.